lib/core/kernel.nit

```diff
@@ -16,7 +16,13 @@
 # This module is the root of the module hierarchy.
 # It provides a very minimal set of classes and services used as a
 # foundation to define other classes and methods.
-module kernel
+module kernel is
+	new_annotation test
+	new_annotation before
+	new_annotation before_all
+	new_annotation after
+	new_annotation after_all
```

Comment:
I suggest that you just add them to the primitive annotation list in `check_annotations`